vllm/worker/model_runner.py

```diff
@@ -1564,6 +1564,13 @@ def execute_model(
         # set to None. Thus, if we encounter None for input_tokens, we just
         # skip sampling and return an empty outputs.
         if model_input.input_tokens is None:
+            if not self.is_driver_worker:
```

Comment:
So this is essential right after `compute_logits` and before the sampling logic, right?

Reply:
yes